src/components/JourneySelector.tsx

```diff
@@ -1,57 +1,80 @@
 import React, { useState } from 'react';
-import '../App.css';
-import { fetchFares } from '../helpers/ApiCallHelper';
-import { DepartureInfo, departureInfoFromOutboundJourney } from '../models/DepartureInfo';
-import { FaresAPIResponseJSON } from '../models/FaresAPIResponse';
+import { Alert, Button, LinearProgress, Stack } from '@mui/material';
+import Container from '@mui/material/Container';
+import { getDepartureInfoFromFares } from '../helpers/ApiResponseHelper';
+import { DepartureInfo } from '../models/DepartureInfo';
+import { areStationsEqual, StationModel } from '../models/StationModel';
 import DropdownList from './DropdownList';
+import JourneyDisplayTable from './JourneyDisplayTable';
+import StationListItem from './StationListItem';
 
-const JourneySelector: React.FC = () => {
-    const stations = [
-        { crs: 'KGX', name: 'King\'s Cross' }, 
-        { crs: 'SVG', name: 'Stevenage' }, 
-        { crs: 'YRK', name: 'York' }, 
-        { crs: 'DAR', name: 'Darlington' }, 
-        { crs: 'NCL', name: 'Newcastle' },
-    ];
+const JourneySelector: React.FC =  () => {
+    const stationsMap: Map<string, StationModel> = new Map();
+    stationsMap.set('KGX', { crs: 'KGX', name: 'King\'s Cross' }); 
+    stationsMap.set('SVG', { crs: 'SVG', name: 'Stevenage' }); 
+    stationsMap.set('YRK', { crs: 'YRK', name: 'York' });
+    stationsMap.set('DAR', { crs: 'DAR', name: 'Darlington' });
+    stationsMap.set('NCL', { crs: 'NCL', name: 'Newcastle' });
 
-    const [departureValue, setDepartureValue] = useState(stations[0]);
-    const [arrivalValue, setArrivalValue] = useState(stations[0]);
-
-    const journeys: DepartureInfo[] = [];
+    const [departureStation, setDepartureStation] = useState<StationModel|undefined>( undefined );
+    const [arrivalStation, setArrivalStation] = useState<StationModel|undefined>( undefined );
+    const [journeys, setJourneys] = useState<DepartureInfo[]>([]);
+    const [isSearching, setIsSearching] = useState(false);
+    const [showAlert, setShowAlert] = useState(false);
 
     const searchFares = async () => {
-        const response = await fetchFares({ 
-            originStationCrs: departureValue.crs,
-            destinationStationCrs: arrivalValue.crs,
-            outboundDateTime: new Date(), 
-        }).catch(err => {alert(err); return null;});
-
-        if (!response || response.status != 200) {
-            return;
+        if (departureStation && arrivalStation) {
+            if (areStationsEqual(arrivalStation, departureStation)) {
+                setShowAlert(true);
+            } 
+            else {
+                setShowAlert(false);
+                setJourneys((await getDepartureInfoFromFares(departureStation, arrivalStation)) ?? []);
```

Comment:
not too different but you can directly update the usestate so you don't need the else, something like

```
            const stationsEqual = areStationsEqual(arrivalStation, departureStation);
            setShowAlert(stationsEqual);
            if (stationsEqual) {
                setJourneys((await getDepartureInfoFromFares(departureStation, arrivalStation)) ?? []);
            } 
```